File: An 2/Semestrul 1/Metode Avansate de Programare/Lab 9/lab9_map/lab9_map/TaskContainerFactory.cs
namespace lab9_map;

public class TaskContainerFactory : Factory
{
    private static readonly TaskContainerFactory instance = new TaskContainerFactory();
    
    private TaskContainerFactory() { }
    
    public static TaskContainerFactory Instance
    {
        get
        {
            return instance;
        }
    }
    
    public Container CreateContainer(Strategy strategy)
    {
        if (strategy == Strategy.FIFO)
        {
            //return new QueueContainer();
        }
        else if (strategy == Strategy.LIFO)
        {
            return new StackContainer();
        }
        return null;
    }
}

File: An 2/Semestrul 1/Metode Avansate de Programare/Lab 2/Sem1-MAP/Sem1-MAP/src/TaskContainerFactory.java
public class TaskContainerFactory implements Factory{
    private static TaskContainerFactory instance = new TaskContainerFactory();

    private TaskContainerFactory() {}

    public static TaskContainerFactory getInstance() {
        return instance;
    }

    @Override
    public Container createContainer(Strategy strategy) {
        switch(strategy){
            case FIFO:
                return new T_QueueContainer();
            case LIFO:
                return new StackContainer();
            default: return null;
        }
    }
}
